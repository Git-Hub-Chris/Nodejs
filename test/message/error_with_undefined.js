'use strict';

require('../common');

function test() {
	let a = 'abc ';
	let b = 1;
	let c = ' def';
	let msg = a + b.value + c; //b.value is undefined value
	console.error(msg);
	throw new Error(msg);
}
Object.defineProperty(test, 'name', {value: 'fun ' + undefined + ' tion'});

test();
