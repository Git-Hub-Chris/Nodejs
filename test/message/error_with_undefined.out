abc undefined def
*error_with_undefined.js:*
  throw new Error(msg);
  ^

Error: abc undefined def
    at fun undefined tion (*error_with_undefined.js:*:*)
    at Object.<anonymous> (*error_with_undefined.js:*:*)
    at Module._compile (internal/modules/cjs/loader.js:*:*)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:*:*)
    at Module.load (internal/modules/cjs/loader.js:*:*)
    at Function.Module._load (internal/modules/cjs/loader.js:*:*)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:*:*)
    at internal/main/run_main_module.js:*:*
